headerfs/store.go

```diff
@@ -655,6 +656,26 @@ func NewFilterHeaderStore(filePath string, db walletdb.DB,
 		return fhs, nil
 	}
 
+	// If we have a state assertion then we'll check it now to see if we
+	// need to modify our filter header files before we proceed.
+	if headerStateAssertion != nil {
+		assertedHeader, err := fhs.FetchHeaderByHeight(
+			headerStateAssertion.Height,
+		)
+		if err != nil {
+			return nil, err
```

Comment:
Not sure how we intend to deploy this flag within `lnd`, but if this is set for new clients whom have yet to fetch this filter header from the network then they won't be able to start. In that case, seems like we'd want to keep the assertion reference around in the `blockManager` as well?